### ISSUE TITLE
[HR] Toggle Level Up membership or Ultimate Membership

### ISSUE THREAD

**Issue by colwynyeun** [Prodigy-Hacking/ProdigyMathGameHacking#1174]
**Describe your hack:**

For example I dont want to buy the membership and yea.
**(Optional) Additional information**

**Have you made sure this hack isn't available yet?* (Yes/No)**

**Comment by 32n** [comment]
Isn't this hack already implemented? https://github.com/Prodigy-Hacking/ProdigyMathGameHacking/blob/a110d33279a78f194def2fd4c540e9c3ed0a1092/willsCheatMenu/src/hacks/player.ts#L80

**Comment by 32n** [comment]
If this toggles membership on/off based on if you already have it or not, a membership might not be possible to hack because members would probably be tracked in prodigy's backend.

**Comment by ArcerionDev** [comment]
> Isn't this hack already implemented?
> 
> https://github.com/Prodigy-Hacking/ProdigyMathGameHacking/blob/a110d33279a78f194def2fd4c540e9c3ed0a1092/willsCheatMenu/src/hacks/player.ts#L80

prodigy decided to make 2 subscription types; level up and ultimate

we want to figure out how to get both of them

**Comment by hostedposted** [comment]
I think I have something
```js
_.instance.prodigy.gameContainer.get("98e-4ed0").data.egInfo.tier = 2
```
for level up and
```js
_.instance.prodigy.gameContainer.get("98e-4ed0").data.egInfo.tier = 3
```
for ultimate.

However I don't know how to test it.

**Comment by ArcerionDev** [comment]
this works

however you also need to execute 
```js
_.instance.prodigy.gameContainer.get("98e-4ed0").data.membership.active = true
```

**Comment by ArcerionDev** [comment]
but it doesn't seem to affect the epics page
![image](https://user-images.githubusercontent.com/60945964/140626176-4a0f0def-9e6e-4c79-9c0e-c872f41db354.png)

**Comment by ConnorAhern** [comment]
> I think I have something
> 
> ```js
> _.instance.prodigy.gameContainer.get("98e-4ed0").data.egInfo.tier = 2
> ```
> 
> for level up and
> 
> ```js
> _.instance.prodigy.gameContainer.get("98e-4ed0").data.egInfo.tier = 3
> ```
> 
> for ultimate.
> 
> However I don't know how to test it.

i believe that tier 1 is for the legacy membership, from teacher sharing

**Comment by ConnorAhern** [comment]
> but it doesn't seem to affect the epics page ![image](https://user-images.githubusercontent.com/60945964/140626176-4a0f0def-9e6e-4c79-9c0e-c872f41db354.png)

it might think you have that legacy membership thing

**Comment by ArcerionDev** [comment]
tier one is nothing

**Comment by ArcerionDev** [comment]
that's prodigy basic; i.e. no benefits

**Comment by ConnorAhern** [comment]
ohh i see

**Comment by ArcerionDev** [comment]
pretty sure that ultimate is the legacy or previous membership; level up is something in between

**Comment by ArcerionDev** [comment]
not 100% sure; need to look at the diff

**Comment by ArcerionDev** [comment]
![image](https://user-images.githubusercontent.com/60945964/140626469-973e1aae-258e-451e-865a-6a3d732c8a59.png)

nevermind; level up is legacy membership; ultimate is just this stupid exclusive item thing

**Comment by ArcerionDev** [comment]
this won't be needed as uhhh we can hack items anyway

**Comment by ConnorAhern** [comment]
> ![image](https://user-images.githubusercontent.com/60945964/140626469-973e1aae-258e-451e-865a-6a3d732c8a59.png)
> 
> nevermind; level up is legacy membership; ultimate is just this stupid exclusive item thing

no, you're incorrect

legacy membership = the membership before they added tiers

**Comment by ConnorAhern** [comment]
i know because im from the wiki and the wiki knows all

**Comment by ConnorAhern** [comment]
> this won't be needed as uhhh we can hack items anyway

arc i want to be able to play through the magmayhem quest for the wiki yt please do this please i can bribe glace into doing sus stuff with you if you do it pleaseee

**Comment by ArcerionDev** [comment]
> > ![image](https://user-images.githubusercontent.com/60945964/140626469-973e1aae-258e-451e-865a-6a3d732c8a59.png)
> > nevermind; level up is legacy membership; ultimate is just this stupid exclusive item thing
> 
> no, you're incorrect
> 
> legacy membership = the membership before they added tiers if you didnt upgrade to either level up or premium

yes; level up is the pre-tier membership

**Comment by ArcerionDev** [comment]
> > this won't be needed as uhhh we can hack items anyway
> 
> arc i want to be able to play through the magmayhem quest for the wiki yt please do this please i can bribe glace into doing sus stuff with you if you do it pleaseee

I know; I was mainly thinking about that because prodigy checks your membership status when loading your inbox which I'm assuming is where pets/items will be given

I'm making member toggle to the ultimate membership either way

**Comment by ConnorAhern** [comment]
> > > this won't be needed as uhhh we can hack items anyway
> > 
> > 
> > arc i want to be able to play through the magmayhem quest for the wiki yt please do this please i can bribe glace into doing sus stuff with you if you do it pleaseee
> 
> I know; I was mainly thinking about that because prodigy checks your membership status when loading your inbox which I'm assuming is where pets/items will be given
> 
> I'm making member toggle to the ultimate membership either way

ok

**Comment by ConnorAhern** [comment]
> > > ![image](https://user-images.githubusercontent.com/60945964/140626469-973e1aae-258e-451e-865a-6a3d732c8a59.png)
> > > nevermind; level up is legacy membership; ultimate is just this stupid exclusive item thing
> > 
> > 
> > no, you're incorrect
> > legacy membership = the membership before they added tiers if you didnt upgrade to either level up or premium
> 
> yes; level up is the pre-tier membership

i know that but like for some reason the people who had the pre-tier membership before level up and premium were differentiated their membership is poo poo now and they have to pay an extra $2/month to get things to work again

**Comment by colwynyeun** [comment]
> > but it doesn't seem to affect the epics page ![image](https://user-images.githubusercontent.com/60945964/140626176-4a0f0def-9e6e-4c79-9c0e-c872f41db354.png)
> 
> it might think you have that legacy membership thing

Yes, It doesn't work... If people want mythical epics non hacked well we need to add to toggle the ultimate or level up membership

**Comment by colwynyeun** [comment]
or maybe people dont want to spend money on the ultimate membership
